SalesforceCommerceCloud/pwa-kit
# [Hooks] Fix token refresh - @W-12704554@

We have `pendingToken` in our auth logic that was meant to be temporary. But we did not remove it when no longer pending.

Todos:
- [x] write a new test

How to test:
1. Load http://localhost:3000/
2. Remove the access token from your local storage.. for example`RefArchGlobal_access_token`.
3. Navigate to another page
4. Verify in the browser's Network tab that there is a request to the token endpoint, and all the other SCAPI requests are successful.

# Types of Changes


> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
